components/accelerator/nvidia/error/xid/helper.go

```diff
@@ -0,0 +1,91 @@
+package xid
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/leptonai/gpud/components"
+	"github.com/leptonai/gpud/components/common"
+	"github.com/leptonai/gpud/log"
+)
+
+// StateUpdateBasedOnEvents resolves the state of the XID error component.
+// note: assume events are sorted by time in ascending order
+func StateUpdateBasedOnEvents(events []components.Event) components.State {
+	var xidRebootMap = make(map[uint64]int)
+	var stateExtraInfo map[string]string
+	var suggestedAction *common.SuggestedActions
+	var xidErr *XidError
+	rebootThreshold := 2
+	healthy := true
+	health := components.StateHealthy
+	stateError := ""
+	for _, event := range events {
+		if event.Name == EventNameErroXid {
+			var currXidErr XidError
+			if err := json.Unmarshal([]byte(event.ExtraInfo[EventKeyErroXidData]), &currXidErr); err != nil {
+				log.Logger.Errorf("failed to unmarshal event %s %s extra info: %s", event.Name, event.Message, err)
+				continue
+			}
+			if event.Type == components.EventTypeCritical {
+				if health == components.StateUnhealthy {
```

Comment:
we can define number values for these states that can make the implementation more simple.
For example:
```go
const(
     StateHealthy = 0
     StateDegraded = 1
     StateUnhealthy = 2
)
```

So we can do this:
```go
state := StateHealthy
if event.Type == components.EventTypeCritical {
   state = StateDegraded
}
if state <= currentState {
   // TODO: maybe they have different suggested actions, so we need to check it...
   continue
}
// get the latest suggested action
```